src/entities/listeners.py

```diff
@@ -0,0 +1,57 @@
+from sqlalchemy import Connection, Table, event, inspect
+from sqlalchemy.orm import Mapper
+
+from .models.dao import Base, FinancialInstitutionDao
+from entities.engine.engine import engine
+
+
+def _setup_fi_history(fi_history: Table, mapping_history: Table):
+    def _insert_history(
+        mapper: Mapper[FinancialInstitutionDao], connection: Connection, target: FinancialInstitutionDao
+    ):
+        new_version = target.version + 1 if target.version else 1
```

Comment:
With the alembic doing server_default 1 and the DAO having a default of 0 is there an instance where target.version would not be populated?

Reply:
yes, the dto to dao transformation